linkage-monitor/src/main/java/com/google/cloud/tools/dependencies/linkagemonitor/LinkageMonitor.java

```diff
@@ -16,22 +16,91 @@
 
 package com.google.cloud.tools.dependencies.linkagemonitor;
 
+import com.google.cloud.tools.opensource.classpath.LinkageChecker;
+import com.google.cloud.tools.opensource.classpath.SymbolProblem;
+import com.google.cloud.tools.opensource.dependencies.Bom;
+import com.google.cloud.tools.opensource.dependencies.RepositoryUtility;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Sets;
+import java.io.IOException;
+import org.eclipse.aether.RepositoryException;
+import org.eclipse.aether.RepositorySystem;
+import org.eclipse.aether.RepositorySystemSession;
+import org.eclipse.aether.artifact.Artifact;
+import org.eclipse.aether.resolution.VersionRangeRequest;
+import org.eclipse.aether.resolution.VersionRangeResolutionException;
+import org.eclipse.aether.resolution.VersionRangeResult;
+
 /**
  * Linkage Monitor detects new linkage errors caused by locally-installed snapshot artifacts for a
  * BOM (bill-of-materials).
  */
 public class LinkageMonitor {
 
-  public static void main(String[] arguments) {
+  public static void main(String[] arguments) throws RepositoryException, IOException {
     if (arguments.length < 1) {
       System.err.println(
           "Please specify BOM coordinates. Example: com.google.cloud:libraries-bom:1.2.1");
       System.exit(1);
     }
     String bomCoordinates = arguments[0];
-    System.out.println("Linkage Monitor for " + bomCoordinates);
-    // TODO(#681): Run Linkage Checker for the BOM specified in argument
-    // TODO(#682): Copy the BOM with locally-installed snapshot versions
-    // TODO(#683): Display new linkage errors caused by snapshot versions if any
+    Bom baseline = RepositoryUtility.readBom(bomCoordinates);
+    Bom snapshot = copyWithSnapshot(baseline);
+
+    ImmutableSet<SymbolProblem> problemInBaseline =
+        LinkageChecker.create(baseline).findSymbolProblems().keySet();
+    ImmutableSet<SymbolProblem> problemsInSnapshot =
+        LinkageChecker.create(snapshot).findSymbolProblems().keySet();
+
+    if (problemInBaseline.containsAll(problemsInSnapshot)) {
+      // No new symbol problems introduced by snapshot BOM. Returning success.
+      return;
+    } else {
+      // TODO(#683): Display new linkage errors caused by snapshot versions if any
+      System.err.println("There are one or more new new linkage errors in snapshot versions:");
+      System.err.println(Sets.difference(problemInBaseline, problemsInSnapshot));
```

Comment:
Don't you want this the other way around?

```suggestion
      System.err.println(Sets.difference(problemInSnapshot, problemsInBaseline));
```

And what about using that difference in the condition too?

```java
Set<SymbolProblem> newProblems = Sets.difference(problemsInSnapshot, problemsInBaseline);
if (!newProblems.isEmpty()) {
  // report new problems and exit 1
}
```

Reply:
Now it's like this. 
```
      System.err.println("There are one or more new new linkage errors in snapshot versions:");
      Set<SymbolProblem> errors =
          Sets.filter(problemsInSnapshot, item -> !problemsInBaseline.contains(item));

```

Reply:
Sets.difference(set1, set2)

> The returned set contains all elements that are contained by set1 and not contained by set2.
https://google.github.io/guava/releases/27.1-android/api/docs/com/google/common/collect/Sets.html#difference-java.util.Set-java.util.Set-